libraries/botbuilder-ai/lib/luisRecognizer.js

```diff
@@ -19,36 +18,170 @@ class LuisRecognizer extends botbuilder_1.IntentRecognizer {
+            let recognizerResult = {
+                text: result.query,
+                intents: $this.getIntents(result.intents),
+                entities: entitiesAndMetadata.entities
             };
+            if (verbose) {
+                recognizerResult.$instance = {
+                    entities: entitiesAndMetadata.metadata
+                };
+            }
+            return recognizerResult;
+        });
+    }
+    getIntents(intents) {
+        if (intents) {
+            return intents.reduce((prev, curr) => {
+                prev[curr.intent || ''] = curr.score;
+                return prev;
+            }, {});
+        }
+    }
+    getEntitiesAndMetadata(entities, compositeEntities, verbose) {
+        let $this = this;
+        let entitiesAndMetadata = {
+            entities: {},
+            metadata: {}
+        };
+        let compositeEntityTypes = [];
+        // We start by populating composite entities so that entities covered by them are removed from the entities list
+        if (compositeEntities) {
+            compositeEntityTypes = compositeEntities.map(compositeEntity => compositeEntity.parentType);
+            compositeEntities.forEach(compositeEntity => {
+                entities = $this.populateCompositeEntity(compositeEntity, entities, entitiesAndMetadata, verbose);
+            });
+        }
+        entities.forEach(entity => {
+            // we'll address composite entities separately
+            if (compositeEntityTypes.indexOf(entity.type) > -1)
+                return;
+            this.addProperty(entitiesAndMetadata.entities, entity.type, $this.getEntityValue(entity));
+            if (verbose) {
+                this.addProperty(entitiesAndMetadata.metadata, entity.type, $this.getEntityMetadata(entity));
+            }
+        });
+        return entitiesAndMetadata;
+    }
+    getEntityValue(entity) {
+        if (entity.type.startsWith("builtin.datetimeV2.")) {
+            return entity.resolution && entity.resolution.values && entity.resolution.values.length ?
+                entity.resolution.values[0].timex :
+                entity.resolution;
+        }
+        else if (entity.resolution) {
+            if (entity.type.startsWith("builtin.number")) {
+                return Number(entity.resolution.value);
+            }
+            else {
+                return Object.keys(entity.resolution).length > 1 ?
+                    entity.resolution :
+                    entity.resolution.value ?
+                        entity.resolution.value :
+                        entity.resolution.values;
+            }
+        }
+        else {
+            return entity.entity;
+        }
+    }
+    getEntityMetadata(entity) {
+        return {
+            $startIndex: entity.startIndex,
+            $endIndex: entity.endIndex,
+            $entity: entity.entity,
+            $score: entity.score,
+            $resolution: entity.resolution ?
```

Comment:
In the general case, I don't think we need this field.  It should be in the main JSON rather than duplicated here.  The one case I can think of where you might want to put it here would be the future/past date times, but I'm not sure that is worth preserving.